src/live_restore/live_restore_fs.c

```diff
@@ -1207,109 +1071,105 @@ __wt_live_restore_fh_import_extents_from_string(
+    if (lr_fh_meta->bitmap_size != 0) {
+        /* We shouldn't be importing a bitmap if the live restore has finished. */
+        WT_ASSERT(session, !lr_fh->destination.back_pointer->finished);
+        /* Import a pre-existing bitmap. */
+        __wt_verbose_debug1(session, WT_VERB_LIVE_RESTORE, "%s metadata bitmap: %s", fh->name,
+          lr_fh_meta->bitmap_str);
+        WT_ERR(__live_restore_decode_bitmap(
+          session, lr_fh_meta->bitmap_str, lr_fh_meta->bitmap_size, lr_fh));
     } else {
-        __wt_readlock(session, &lr_fh->ext_lock);
-        __wt_verbose_debug3(session, WT_VERB_LIVE_RESTORE, "%s metadata extent list string: %s",
-          fh->name, extent_str);
-        /* The extents are separated by ;. And have the shape %d-%u. */
-        wt_off_t off = 0, next_off;
-        size_t len;
-        WTI_LIVE_RESTORE_HOLE_NODE **current = &lr_fh->destination.hole_list_head;
-        const char *str_ptr = extent_str;
-        char *next;
-        while (true) {
-            if (!__wt_isdigit((u_char)*str_ptr))
-                WT_ERR_MSG(session, EINVAL, "Invalid offset found in extent string");
-
-            next_off = (wt_off_t)strtoll(str_ptr, &next, 10);
-            str_ptr = next;
-            if (*str_ptr == '\0')
-                WT_ERR_MSG(session, EINVAL, "Invalid separator found in extent string");
-
-            /*
-             * Extents are additive to compress the string size i.e. the offset of extent n + 1 is
-             * the offset of extent n plus the offset of extent n + 1.
-             */
-            off += next_off;
-            str_ptr++;
-            if (!__wt_isdigit((u_char)*str_ptr))
-                WT_ERR_MSG(session, EINVAL, "Invalid length found in extent string");
-
-            len = (size_t)strtol(str_ptr, &next, 10);
-            if (len == 0)
-                WT_ERR_MSG(session, EINVAL, "Length zero extent found, this is an error");
-
-            __wt_verbose_debug3(session, WT_VERB_LIVE_RESTORE,
-              "Adding an extent: %" PRId64 "-%" WT_SIZET_FMT, off, len);
-            WT_ERR(__live_restore_alloc_extent(session, off, len, NULL, current));
-            current = &((*current)->next);
-
-            str_ptr = next;
-            /* We've reached the end of the string, don't go over by accident. */
-            if (*str_ptr == '\0')
-                break;
-            str_ptr++;
+        if (lr_fh->source_size == 0)
+            lr_fh->destination.complete = true;
+        else {
+            if (lr_fh->destination.newly_created) {
+                /*
+                 * Only allocate a bitmap for a file that has been created in the destination for
+                 * this run.
+                 */
+                uint64_t bitmap_size = lr_fh->source_size / lr_fh_meta->allocsize;
+                WT_ERR(__bit_alloc(session, bitmap_size, &lr_fh->destination.bitmap));
+                lr_fh->destination.bitmap_size = bitmap_size;
```

Comment:
I don't think this necessarily relies on the outer if blocks, can we move it above and return early from the function in the case of newly created files?

Reply:
Great suggestion! This is a lot cleaner: eea23811e. I pushed it up, lets see how it goes with testing.